packages/ui-lib/src/Form/atoms/Switch.tsx

```diff
@@ -294,7 +305,7 @@ const Switch : React.FC<IProps> = ({ state = 'default', leftTheme = 'off', right
         </SwitchInner>
       </SwitchOuter>
       {labelText ? <LabelText>{labelText}</LabelText> : null}
-      <RealInput ref={inputRef} type='checkbox' disabled={state !== 'default' && state !== 'failure'} {...(isDefaultPropPassed ? { defaultChecked } : { defaultChecked: checked })} /> 
+      <RealInput ref={inputRef} type='checkbox' disabled={state !== 'default' && state !== 'failure'} defaultChecked={checked} />
```

Comment:
Although this preserves the usage of the previous implementation. I think this line needs to be updated, now that we are adding a defaultCheck value to the component. If the user sends a defaultCheck value this should be assigned instead of the checked value.

I'm suggesting

defaultChecked={defaultChecked?  defaultChecked : checked}

I think this way it won't become a breaking change but will improve the implementation. Please let me know your thoughts in this part of the implementation.